src/main/java/ua/com/juja/microservices/keepers/slackbot/controller/KeepersSlackCommandController.java

```diff
@@ -1,23 +1,70 @@
 import org.slf4j.LoggerFactory;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.*;
+import ua.com.juja.microservices.keepers.slackbot.model.SlackParsedCommand;
+import ua.com.juja.microservices.keepers.slackbot.model.request.KeeperRequest;
 import ua.com.juja.microservices.keepers.slackbot.service.KeeperService;
 import ua.com.juja.microservices.keepers.slackbot.service.impl.SlackNameHandlerService;
 
+import java.util.Arrays;
+
 /**
  * @author Nikolay Horushko
+ * @author Dmitriy Lyashenko
  */
 @RestController
 public class KeepersSlackCommandController {
     private final Logger logger = LoggerFactory.getLogger(this.getClass());
 
+    @Value("${slack.slashCommandToken}")
+    private String slackToken;
+
     private KeeperService keeperService;
     private SlackNameHandlerService slackNameHandlerService;
 
     public KeepersSlackCommandController(KeeperService keeperService, SlackNameHandlerService slackNameHandlerService) {
         this.keeperService = keeperService;
         this.slackNameHandlerService = slackNameHandlerService;
     }
+
+    @PostMapping(value = "/commands/keeper-add", consumes = MediaType.APPLICATION_FORM_URLENCODED_VALUE)
+    public RichMessage onReceiveSlashCommandKeeperAdd(@RequestParam("token") String token,
+                                                     @RequestParam("user_name") String fromUser,
+                                                     @RequestParam("text") String text) {
+
+        logger.debug("Received slash command KeeperAdd: from user: [{}] command: [{}] token: [{}]",
+                fromUser, text, token);
+
+        if (!token.equals(slackToken)) {
+            logger.warn("Received invalid slack token: [{}] in command KeeperAdd for user: [{}]", token, fromUser);
+            return new RichMessage("Sorry! You're not lucky enough to use our slack command.");
+        }
+
+        String response = "ERROR. Something went wrong and we didn't award the users :(";
+
+        logger.debug("Started create slackParsedCommand and create keeper request");
+        SlackParsedCommand slackParsedCommand = slackNameHandlerService.createSlackParsedCommand(fromUser, text);
+        KeeperRequest keeperRequest = new KeeperRequest(slackParsedCommand);
+        logger.debug("Finished create slackParsedCommand and create keeper request");
+
+        logger.debug("Sent keeper request to Keeper service. KeeperRequest: [{}]",
+                keeperRequest.toString());
+        String[] result = keeperService.sendKeeperAddRequest(keeperRequest);
+        logger.debug("Received response from Keeper service: [{}]", Arrays.toString(result));
+
+        if (result.length > 0) {
+            response = "Thanks, we added a new Keeper " + result[0]
+                    + " in direction {" + keeperRequest.getDirection() + "}";
+        }
```

Comment:
It is a wrong way to use result[0], because according with keeper documentation
 https://github.com/JujaLabs/docs/tree/master/architecture/keepers KPR-F1-RSP
keeper return id of saved request. it is not a slackName!
Sample solution you can see in the gamification-slack-bot for 'thanks' or 'codenjoy' achievement or do better.